src/helpers.c

```diff
@@ -579,7 +612,44 @@ enum swupd_code swupd_init(enum swupd_init_config config)
 	}
 
 	if ((config & SWUPD_NO_NETWORK) == 0) {
-		if (swupd_curl_init() != 0) {
+
+		bool curl_init = true;
+
+		/* try to init curl with whatever server is set up */
+		if (swupd_curl_init(NULL) != 0) {
+			/* the server is not responding */
+			if (mirror_is_set()) {
+				/* the server is a mirror, check to see if the upstream server is working */
+				swupd_curl_deinit();
+				char *upstream_version_url = NULL;
+				get_upstream_version_url(&upstream_version_url);
+				if (upstream_version_url && swupd_curl_init(upstream_version_url) == 0) {
+					/* the upstream server is up, unset the mirror */
+					warn("The mirror '%s' is not responding\n", version_url);
```

Comment:
If the mirror is offline we should hard fail, so we don't need to unset_mirror_url() here.
If the mirror is online, but outdated (or errors on downloading last version) we should unset the mirror and continue

Reply:
true, I'll fix that

Reply:
That is exactly how it was before... 😖 I guess there was a confusion somewhere